agentos/registry.py

```diff
@@ -0,0 +1,117 @@
+        return component
+
+    def _clone_repo(self, repo_name):
+        repo = self.registry["repos"][repo_name]
+        assert repo["type"] == "github", f'Uknown repo type: {repo["type"]}'
+        org_name, proj_name = repo["url"].split("/")[-2:]
+        clone_destination = self.aos_cache_dir / org_name / proj_name
+        if not clone_destination.exists():
+            cmd = ["git", "clone", repo["url"], clone_destination]
+            result = subprocess.run(cmd)
+            assert result.returncode == 0, f"Git clone non-zero return: {cmd}"
+        assert clone_destination.exists(), f"Unable to clone {repo['url']}"
+        return clone_destination
+
+    def _checkout_version(self, repo_path, version_string):
+        curr_dir = os.getcwd()
+        os.chdir(repo_path)
+        cmd = ["git", "checkout", "-q", version_string]
+        result = subprocess.run(cmd)
+        assert result.returncode == 0, f"FAILED: {cmd} in {repo_path}"
+        os.chdir(curr_dir)
+
+
+if __name__ == "__main__":
+    params = {
+        "acme_r2d2_agent": {
+            "evaluate": {"num_episodes": 50},
+            "learn": {"num_episodes": 100},
+        },
+        "acme_r2d2_dataset": {
+            "__init__": {
+                "batch_size": 32,
+                "discount": 0.99,
+                "max_priority_weight": 0.9,
+                "max_replay_size": 500,
+                "priority_exponent": 0.6,
+                "replay_period": 40,
+                "sequence_length": 13,
+                "store_lstm_state": True,
+            }
+        },
+        "acme_cartpole": {
+            "__init__": {
+                "batch_size": 32,
+                "discount": 0.99,
+                "max_replay_size": 500,
+                "replay_period": 40,
+                "sequence_length": 13,
+                "store_lstm_state": True,
+            }
+        },
+        "acme_r2d2_policy": {
+            "__init__": {
+                "batch_size": 32,
+                "discount": 0.99,
+                "epsilon": 0.01,
+                "max_replay_size": 500,
+                "replay_period": 40,
+                "sequence_length": 13,
+                "store_lstm_state": True,
+            }
+        },
+        "acme_r2d2_trainer": {
+            "__init__": {
+                "adam_epsilon": 0.001,
+                "batch_size": 32,
+                "burn_in_length": 2,
+                "clip_grad_norm": None,
+                "discount": 0.99,
+                "importance_sampling_exponent": 0.2,
+                "learning_rate": 0.001,
+                "max_replay_size": 500,
+                "min_replay_size": 50,
+                "n_step": 5,
+                "replay_period": 40,
+                "samples_per_insert": 32.0,
+                "sequence_length": 13,
+                "store_lstm_state": True,
+                "target_update_period": 20,
+            }
+        },
+    }
+
+    mlflow.start_run()
+    registry = Registry()
```

Comment:
What do you think of having the functionality of the registry be at the module-level so that a developer doesn't have to create an instance of an `agentos.Registry` object if they are just interacting with the definitive registry (vs. e.g., trying to talk to a private registry set up by their organization)

This would just save the developer from having to learn/remember one more step when interacting with the definitive registry.

There are a couple ways we could do this: 

1. Still have a `Registry` class and make a module-level instance of the class (e.g., `agentos.registry._agentos_registry`) and add module-level wrapper functions for get_component and publish_component, etc.
2. Remove the Registry class altogether and just have all of the functionality at the module-level. If we go down this path, we could still support a future where there are many unofficial registries (e.g. private registries hosted by an org) by adding module-level functions like `reconnect(new_registry_url)`.

I'm leaning towards # 1, but only slightly

In either case, what do you think of adding aliases to the root agentos module itself (via [the __all__ dunder](https://github.com/agentos-project/agentos/blob/262fa003fc8dbacdcca7555050179c578334384b/agentos/__init__.py#L16)), to support usage like:

```python
import agentos
acme_r2d2 = agentos.get_component("acme_r2d2")
```